src/PhpSpec/CodeGenerator/Generator/Argument/StringBuilder.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace PhpSpec\CodeGenerator\Generator\Argument;
+
+class StringBuilder
+{
+    /**
+     * @param \ReflectionParameter[] $parameters
+     *
+     * @return string
+     */
+    public function buildFromReflectionParameters(array $parameters, string $targetClassNamespace) : string
+    {
+        return implode(', ', array_map(function (\ReflectionParameter $parameter) use ($targetClassNamespace) {
+            return $this->buildArgument($parameter, $targetClassNamespace);
+        }, $parameters));
+    }
+
+    private function buildArgument(\ReflectionParameter $parameter, string $targetClassNamespace) : string
+    {
+        $parameterName = '$' . $parameter->getName();
+
+        $type = $parameter->getType();
+
+        switch (true) {
```

Comment:
This type logic can be extracted to a method, and possibly replaced with if/else

Reply:
I usually use `switch` for > 2 cases, just personal preference - I will change it to an if/else and extract it